### PR TITLE
Fix issue with Max DateTime

### PR DESCRIPTION
**Pull request by JoshLove-msft**

### CODE REVIEW

**Comment by JoshLove-msft** [comment]
azp run net - servicebus - tests

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpMessageConverter.cs [hunk 1]
```diff
@@ -214,6 +222,15 @@ public virtual ServiceBusReceivedMessage AmqpMessageToSBReceivedMessage(AmqpMess
                 annotatedMessage.Header.DeliveryCount = isPeeked ? annotatedMessage.Header.DeliveryCount : annotatedMessage.Header.DeliveryCount + 1;
             }
 
+            if (annotatedMessage.HasSection(AmqpMessageSection.MessageAnnotations) &&
+                annotatedMessage.MessageAnnotations.TryGetValue(AmqpMessageConstants.LockedUntilName, out object lockedUntil))
```
Comment (jsquire) [inline sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpMessageConverter.cs:226]: Just to be sure that I'm following:  

```text
IF we have a section 
    AND we have a value 
    AND that value is greater than the offset max 
THEN
    force the value to the offset max
```

So, if we have a value that isn't too large, then we just use it as-is?

**Review by jsquire — APPROVED** [review]

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpMessageConverter.cs (hunk 1 above)
Comment (JoshLove-msft) [reply sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpMessageConverter.cs:226]: Yep

**Pull request merged**